paasta_tools/tron_tools.py

```diff
@@ -387,11 +389,25 @@ def check_actions(self) -> Tuple[bool, List[str]]:
                 msgs.extend(action_msgs)
         return checks_passed, msgs
 
+    def check_deploy_group(self) -> Tuple[bool, str]:
+        deploy_groups = [action.get_deploy_group() for action in self.get_actions()]
+        deploy_groups.append(self.get_deploy_group())
+        check_deploy_groups = [deploy_group for deploy_group in deploy_groups if deploy_group is not None]
+        if len(check_deploy_groups) > 0:
+            pipeline_steps = [step['step'] for step in get_pipeline_config(self.service, self.soa_dir)]
```

Comment:
the service can also vary by action, so I think you might want to do this verification in the action instead